packages/remix-dev/compiler/plugins/cssModulesPlugin.ts

```diff
@@ -0,0 +1,387 @@
+          path: getResolvedFilePath(config, args),
+          sideEffects: false,
+        };
+      });
+
+      build.onLoad(
+        { filter: suffixMatcher, namespace: pluginNamespace + "-faker" },
+        async (args) => {
+          let res = await assetsManifestPromiseRef.current;
+          let json = res?.cssModules?.moduleMap[args.path].json || {};
+          return {
+            contents: JSON.stringify(json),
+            loader: "json",
+          };
+        }
+      );
+    },
+  };
+}
+
+async function processCssCached({
+  config,
+  filePath,
+}: {
+  config: RemixConfig;
+  filePath: string;
+}): Promise<CssModuleFileContents> {
+  // TODO: Caching as implemented not working as expected in dev mode. Bypass
+  // for now.
+  return await processCss({ config, filePath });
+
+  //   let file = path.resolve(config.appDirectory, filePath);
+  //   let hash = await getFileHash(file);
+  //   // Use an on-disk cache to speed up dev server boot.
+  //   let processedCssPromise = (async function () {
+  //     let key = file + ".cssmodule";
+  //     let cached: (CssModuleFileContents & { hash: string }) | null = null;
+  //     try {
+  //       cached = await cache.getJson(config.cacheDirectory, key);
+  //     } catch (error) {
+  //       // Ignore cache read errors.
+  //     }
+  //     if (!cached || cached.hash !== hash) {
+  //       let processed = await processCss({ config, filePath });
+  //       cached = { hash, ...processed };
+  //       try {
+  //         await cache.putJson(config.cacheDirectory, key, cached);
+  //       } catch (error) {
+  //         // Ignore cache put errors.
+  //       }
+  //     }
+  //     return cached;
+  //   })();
+  //   return processedCssPromise;
+}
+
+export async function processCss({
+  config,
+  filePath,
+}: {
+  config: RemixConfig;
+  filePath: string;
+}): Promise<CssModuleFileContents> {
+  let classPrefix =
+    path.basename(filePath, path.extname(filePath)).replace(/\./g, "-") + "__";
+  let source = await fse.readFile(filePath);
+
+  let res = parcelTransform({
+    filename: path.relative(config.appDirectory, filePath),
+    code: source,
+    cssModules: true,
+    minify: process.env.NODE_ENV === "production",
+    analyzeDependencies: true, // TODO: Maybe?
+    sourceMap: true,
+    drafts: { nesting: true },
+  });
+
+  let json: CssModuleClassMap = {};
+  let cssModulesContent = decoder.decode(res.code);
+  let parcelExports = res.exports || {};
+
+  // sort() to keep order consistent in different builds
+  for (let originClass of Object.keys(parcelExports).sort()) {
+    let props = parcelExports[originClass];
+    let patchedClass = props.name;
+    let prefixedClassName = getPrefixedClassName(classPrefix, patchedClass);
+    json[originClass] = props.composes.length
+      ? getComposedClassNames(classPrefix, prefixedClassName, props.composes)
+      : prefixedClassName;
+    cssModulesContent = cssModulesContent.replace(
```

Comment:
I guess if we added support for something like postcss-modules `generateScopedName` you could avoid this? https://github.com/parcel-bundler/parcel-css/issues/156

Reply:
FYI working on that over here if you have any feedback: https://github.com/parcel-bundler/parcel-css/pull/180